### PR TITLE
feat(codewhisperer): ignore white spaces in right context matching

### PR DESCRIPTION
**Pull request by zixlin7**
## Problem
to ignore white spaces and new lines in right context matching to handle more cases
before: 

https://user-images.githubusercontent.com/60411978/226684454-75798fdb-a5d1-4046-8057-de894de9c8df.mov

## Solution
after:

https://user-images.githubusercontent.com/60411978/226685059-9a2442d2-86c4-4bf0-a9db-4ef7066cf7a1.mov

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by andrewyuq — COMMENTED** [review]
can we add a unit test to this?

File: src/codewhisperer/service/inlineCompletionService.ts [hunk 1]
```diff
@@ -90,14 +90,15 @@ class CWInlineCompletionItemProvider implements vscode.InlineCompletionItemProvi
...
         let rightContextRange: vscode.Range | undefined = undefined
         const pos = vscode.window.activeTextEditor?.selection.active || RecommendationHandler.instance.startPos
-        if (suggestion.split(/\r?\n/).length > 1) {
-            rightContextRange = new vscode.Range(pos, document.positionAt(document.offsetAt(pos) + suggestion.length))
+        const lines = suggestion.split(/\r?\n/).length
+        if (lines > 1) {
+            rightContextRange = new vscode.Range(pos, document.lineAt(pos.line + lines).range.end)
         } else {
-            rightContextRange = new vscode.Range(pos, document.lineAt(pos).range.end)
+            rightContextRange = new vscode.Range(pos, document.lineAt(pos.line + 1).range.end)
```
Comment (leigaol) [inline src/codewhisperer/service/inlineCompletionService.ts:97]: Could you explain why this + 1 is necessary? Thanks!

Comment (zixlin7) [reply src/codewhisperer/service/inlineCompletionService.ts:97]: this is for when I invoke at line1, suggestion is one line and an exact match of my right context at line 2

**Review by andrewyuq — APPROVED** [review]

**Comment by zixlin7** [comment]
> Is this a user-facing change that is worth mentioning in the changelog?

added changelog, thanks

**Pull request merged**